### ISSUE TITLE
dynamically resolve super.foo

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#30000]
```dart
class A { foo() => print(1); }
class B { foo() => super.foo(); } // Note: extends A is missing
class C extends A with B {}
main() => new C().foo();
```

according to the spec, this program:
  - has a static warning in `super.foo` because B doesn't extend A
  - calling `new B().foo()` should produce a runtime noSuchMethod exception
  - calling `new C().foo()` should call `A.foo` and print.

Issue #29758 is tracking whether the warning should be omitted if you change B declaration to `B implements A`. But it's my understanding that the spec says that regardless of the warning, we should have the semantics described here.

Today `new C().foo()` produces a noSuchMethod exception.

The problem appears to be that front-end eagerly tries to handle the situation of `new B().foo` by creating the [noSuchMethod invocation][1], but the mixin transformation copies that code. 

Should we instead only issue a warning in body\_builder and generate code that still attempts to call super.foo?  Later the mixin transformer will copy the code and fix it up  (changing the code into `noSuchMethod` calls if the member is missing in the super class.)

[1]: https://github.com/dart-lang/sdk/blob/3144296c4f5b7f3fd9f2f320d3bc0e2012224b6e/pkg/front_end/lib/src/fasta/kernel/body_builder.dart#L913

**Comment by stereotype441** [comment]
FYI, in https://codereview.chromium.org/2958113002/ I'm adding a test case to make sure type inference won't choke on a class like `class B` above.

**Comment by peter-ahe-google** [comment]
Proposed fix: [CL 2963763002](https://codereview.chromium.org/2963763002/).
